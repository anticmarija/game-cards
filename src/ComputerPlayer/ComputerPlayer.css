.computer-player {
  height: 120px;
  position: absolute;
  padding: 5px;
}

.computer-player-score {
  color: yellowgreen;
  font-size: 24px;
}

.computer-player2 {
  top: 50%;
  left: 0;
  transform: translateY(-50%);
}

.computer-player3 {
  top: 15px;
  left: 50%;
  transform: translateX(-50%);
  padding-left: 40px;
}

.computer-player4 {
  top: 50%;
  right: 0;
  transform: translateY(-50%);
}

.computer-player-img {
  width: 80px;
  height: 120px;
  margin-left: -40px;
}

.computer-player2 .computer-player-img:first-child {
  margin-left: 5px;
}

.computer-player4 .computer-player-img:last-child {
  margin-right: 5px;
}

.computer-player2 .computer-player-img {
  margin-left: -40px;
}

.computer-player4 .computer-player-img {
  margin-right: -40px;
  margin-left: 0;
}

@media only screen and (max-width: 1400px) {
  .computer-player3 {
    padding-left: 60px;
  }

  .computer-player2,
  .computer-player4 {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .computer-player2 .computer-player-img,
  .computer-player4 .computer-player-img {
    margin: -80px 40px 0 40px;
    transform: rotate(90deg);
  }

  .computer-player2 .computer-player-score,
  .computer-player4 .computer-player-score {
    position: absolute;
    top: -75px;
    width: 100%;
  }

  .computer-player2 .computer-player-score {
    left: 150px;
  }

  .computer-player4 .computer-player-score {
    right: 150px;
    text-align: right;
  }

  .computer-player2 .computer-player-img:first-child {
    margin-left: 40px;
  }
}

@media only screen and (max-width: 930px) {
  .computer-player-img {
    margin-left: -60px;
  }

  .computer-player3 {
    padding-left: 60px;
  }

  .computer-player2 .computer-player-img {
    margin-left: -40px;
  }

  .computer-player2 .computer-player-img,
  .computer-player4 .computer-player-img {
    margin: -100px 40px 0 40px;
  }

  .computer-player2 .computer-player-score,
  .computer-player4 .computer-player-score {
    top: -145px;
  }

  .computer-player2 .computer-player-score {
    left: 25px;
  }

  .computer-player4 .computer-player-score {
    right: 25px;
  }

  .computer-player4 .computer-player-img:last-child {
    margin-right: 40px;
  }
}

@media only screen and (max-width: 530px) {
  .computer-player-img {
    width: 40px;
    height: 60px;
    margin-left: -30px;
  }

  .computer-player3 {
    padding-left: 30px;
  }

  .computer-player2 .computer-player-img,
  .computer-player4 .computer-player-img {
    margin: -50px 10px 0 10px;
  }

  .computer-player2 .computer-player-score,
  .computer-player4 .computer-player-score {
    top: -100px;
  }

  .computer-player2 .computer-player-img:first-child {
    margin-left: 10px;
  }

  .computer-player4 .computer-player-img:last-child {
    margin-right: 10px;
  }
}
