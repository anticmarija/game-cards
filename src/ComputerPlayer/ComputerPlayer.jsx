import React from 'react';
import { connect } from 'react-redux';
import './ComputerPlayer.css';
import cardBackground from '../assets/card-background.png';

export const ComputerPlayer = ({ player, playerCardsLength, playerScore }) => {
  return (
    <div className={`computer-player computer-${player}`}>
      <div className="computer-player-score">
        Player {player.substr(-1)}: {playerScore}
      </div>
      {[...Array(playerCardsLength)].map((_, i) => (
        <img
          className="computer-player-img"
          key={i}
          src={cardBackground}
          alt="background"
        />
      ))}
    </div>
  );
};

export const mapStateToProps = (state, ownProps) => {
  return {
    playerCardsLength:
      state.game.players[ownProps.player] &&
      state.game.players[ownProps.player].cards.length,
    playerScore:
      state.game.players[ownProps.player] &&
      state.game.players[ownProps.player].score,
  };
};

export default connect(mapStateToProps)(ComputerPlayer);
