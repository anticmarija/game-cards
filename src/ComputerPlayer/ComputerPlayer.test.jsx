import React from 'react';
import { render } from '@testing-library/react';
import { ComputerPlayer } from './ComputerPlayer';

describe('Test ComputerPlayer component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ComputerPlayer
        player="player2"
        playerCardsLength={10}
        playerScore={0}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
