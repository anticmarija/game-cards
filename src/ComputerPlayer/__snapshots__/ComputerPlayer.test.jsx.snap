// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ComputerPlayer component should render correctly 1`] = `
<div
  class="computer-player computer-player2"
>
  <div
    class="computer-player-score"
  >
    Player 
    2
    : 
    0
  </div>
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
  <img
    alt="background"
    class="computer-player-img"
    src="card-background.png"
  />
</div>
`;
